uploadNeuroDB/NeuroDB/MRI.pm

```diff
@@ -119,71 +121,77 @@ INPUTS:
   - $subjectIDref: hash reference of subject IDs
   - $studyDate   : study date
   - $dbhr        : database handle reference
-  - $objective   : the objective of the study
   - $db          : database object
 
 RETURNS: the session ID of the visit
 
 =cut
 
 sub getSessionID {
-    my ($subjectIDref, $studyDate, $dbhr, $objective, $db) = @_;
+    my ($subjectIDref, $studyDate, $dbhr, $db) = @_;
     my ($sessionID, $studyDateJD);
     my ($query, $sth);
     my $dbh = $$dbhr;
 
-# find a matching timepoint
-    $query = "SELECT ID, Date_visit, Visit FROM session WHERE CandID=$subjectIDref->{'CandID'} AND LOWER(Visit_label)=LOWER(".$dbh->quote($subjectIDref->{'visitLabel'}).") AND Active='Y'";
+    # find a matching timepoint
+    $query = "SELECT ID, Date_visit, Visit "
```

Comment:
shouldn't you select the projet and subproject of the visit here?

Reply:
Since the function is called `getSessionID`, only the session ID is returned. The sub-project ID is computed in method `getObjective`.

Also, I checked on branch `master` and there are no project IDs associated to sessions, only sub-project IDs. AM I missing something. When determining which protocol to use we only consider the **candidate** project ID anyway (i.e. the project to which a candidate is originally registered).

Reply:
OK. Makes sense to select the subproject in `getObjective`. Probably should select the `ProjectID` there too or create a new function for it?

Regarding your last comment about no ProjectID in `session`, I just checked `master` on the LORIS repo and there is a `ProjectID` field in the `session` table (see https://github.com/aces/Loris/blob/12a235e511b810d35c39475db1b48f04fa8e85c9/SQL/0000-00-00-schema.sql#L182) so it might be a good idea to select the `ProjectID` of the session in the MRI scripts. What do you think?